lib/rubygems/defaults.rb

```diff
@@ -59,7 +59,7 @@ def self.default_rubygems_dirs
   # Path to specification files of default gems.
 
   def self.default_specifications_dir
-    File.join(Gem.default_dir, "specifications", "default")
+    @default_specifications_dir ||= File.join(Gem.default_dir, "specifications", "default")
```

Comment:
I think the current spec failure means the monkeypatch in `bundler/spec/support/hax.rb` to fake the default gem dir needs to be adapted now to also modify this new instance variable.

Reply:
This code here

https://github.com/rubygems/rubygems/blob/2db04e0a76e16889a7849f44e5599f28611ec66c/bundler/spec/support/hax.rb#L12

Reply:
Ah, I didn't check the Bundler's test. Thanks for telling me!
Fixed in https://github.com/rubygems/rubygems/pull/4951/commits/19f5148e7e01935c6f93441560519040ca35668b